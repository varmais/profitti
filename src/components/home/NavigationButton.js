import React, { Component, PropTypes } from 'react';
import {
  StyleSheet,
  Text,
  TouchableHighlight
} from 'react-native';
import config from '../../config';

const styles = StyleSheet.create({
  menuText: {
    fontSize: 18,
    color: config.colors.white,
    fontWeight: 'bold',
    alignSelf: 'center'
  },
  menuItem: {
    height: 40,
    backgroundColor: config.colors.green,
    borderColor: config.colors.greenlight,
    borderWidth: 1,
    borderRadius: 8,
    marginBottom: 16,
    justifyContent: 'center'
  }
});

export default class NavigationButton extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    path: PropTypes.string.isRequired,
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired
    }).isRequired
  };

  render() {
    return (
      <TouchableHighlight style={styles.menuItem} underlayColor='#73941f' onPress={this.handlePress}>
        <Text style={styles.menuText}>
          {this.props.title}
        </Text>
      </TouchableHighlight>
    );
  }

  handlePress = () => {
    this.props.navigation.navigate(this.props.path);
  };
}
