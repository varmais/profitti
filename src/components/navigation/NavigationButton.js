import React, { Component, PropTypes } from 'react';
import {
  PixelRatio,
  StyleSheet,
  Text,
  View
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import Touchable from '../common/Touchable';
import config from '../../config';

const styles = StyleSheet.create({
  button: {
    height: 60,
    backgroundColor: config.colors.white,
    borderWidth: 0,
    justifyContent: 'center'
  },
  text: {
    alignSelf: 'flex-start',
    color: config.colors.blackLight,
    fontSize: 18
  },
  textWithIcon: {

  },
  arrow: {
    position: 'absolute',
    right: 0,
    bottom: -4 / PixelRatio.get()
  }
});

export default class NavigationButton extends Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    icon: PropTypes.string,
    onPress: PropTypes.func.isRequired
  };

  render () {
    const { onPress } = this.props;
    return (
      <Touchable style={styles.button} onPress={onPress} underlayColor={config.colors.gray}>
        <View>
          {this.renderIcon()}
          {this.renderText()}
          <Icon name="ios-arrow-forward" size={24} style={styles.arrow} />
        </View>
      </Touchable>
    );
  }

  renderIcon () {
    const { icon } = this.props;
    if (icon) {
      return <View/>
    }
  }

  renderText () {
    const { icon, text } = this.props;
    const textStyles = [styles.text];
    if (icon) {
      textStyles.push(textStyles.textWithIcon);
    }
    return (
      <Text style={textStyles}>
        {text}
      </Text>
    );
  }
}