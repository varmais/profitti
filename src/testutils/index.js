import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

global.React = React;
global.renderer = renderer;

export function createNavigationProp (params = {}) {
  return {
    state: {params: params},
    navigate: () => {}
  }
}