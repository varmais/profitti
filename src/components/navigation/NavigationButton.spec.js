import '../../testutils';
import NavigationButton from './NavigationButton';

describe('components:navigation:NavigationButton', () => {
  let onPressStub;
  const text = 'button text';
  const icon = 'glass';

  beforeEach(() => {
    onPressStub = jest.fn();
  });

  describe('when icon has not been given', () => {
    test('renders NavigationButton', () => {
      const snapshot = renderer.create(<NavigationButton text={text} onPress={onPressStub} />).toJSON();
      expect(snapshot).toMatchSnapshot();
    });

    describe('when button is pressed', () => {
      test('it calls onPress callback');
    });
  });

  describe('when icon is given', () => {
    test('renders NavigationButton', () => {
      const snapshot = renderer.create(<NavigationButton icon={icon} text={text} onPress={onPressStub} />).toJSON();
      expect(snapshot).toMatchSnapshot();
    });
  });
});
