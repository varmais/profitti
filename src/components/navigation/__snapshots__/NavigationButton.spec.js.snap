// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:navigation:NavigationButton when icon has not been given renders NavigationButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderWidth": 0,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#222",
            "fontSize": 18,
          },
        ]
      }
    >
      button text
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "bottom": -2,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`components:navigation:NavigationButton when icon is given renders NavigationButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderWidth": 0,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#222",
            "fontSize": 18,
          },
          Object {
            "paddingLeft": 32,
          },
        ]
      }
    >
      button text
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 18,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "bottom": -2,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
